app/service/url_shortener.py

```diff
@@ -50,6 +50,17 @@ def _create_random_short_url(db, length=8) -> str:
     #return as a string
     return str(url)
 
+def get_redirect_url(short_url, is_redirect: bool =False):
+    original_url = db.get_link(short_url)
```

Comment:
you probably will have this pattern of `link = db.get_link(url)` 

You can rase the ValueError in that function. You can write 2 functions

1. get_link: same as you have it now
2. get_link_or_fail: calls get_link but then has the if/else to return value error if the link is None